start_esp/nginx-auto.conf.template

```diff
@@ -115,19 +115,42 @@ http {
 
 % if cors_preset == 'basic':
     # Begin basic CORS preset
-    if (${'$'}request_method = 'OPTIONS') {
-      add_header 'Access-Control-Max-Age' 1728000;
-      add_header 'Access-Control-Allow-Origin' '${cors_allow_origin}';
-      add_header 'Access-Control-Allow-Methods' '${cors_allow_methods}';
-      add_header 'Access-Control-Allow-Headers' '${cors_allow_headers}';
-      add_header 'Content-Type' 'text/plain; charset=utf-8';
-      add_header 'Content-Length' 0;
-      return 204;
+  % if cors_allow_origin_regex != '':
```

Comment:
With this idea, we have to use preset mode "basic" or "cors_with_regex" to interpret flag "cors_allow_origin" differently.   This is confusing.  I rather use the meaningful flag name to call it out, such as cors_allow_origin and cors_allow_origin_regex.

Reply:
When using the preset mode "cors_with_regex", "cors_allow_origin" is not needed and "cors_allow_origin_regex" can be used to express the allowed origins.

Reply:
So you are proposing:  cors_allow_origin_regex only works for preset = cors_with_regex,  cors_allow_origin only works for preset = basic?    
I think it is confusing.

Reply:
Yes, cors_allow_origin_regex only works under the preset mode 'cors_with_regex', whereas cors_allow_origin only works under the preset mode 'basic'. To avoid confusions, the variable 'cors_allow_origin' can be renamed to 'cors_allow_origin_no_regex'. With this approach, the current 'basic' mode remains to be easy to understand and work for many users.